autoload/zepl.vim

```diff
@@ -100,6 +100,46 @@ function! zepl#send(text) abort
     endif
 endfunction
 
+function! zepl#send_sigint() abort
+  if has('nvim')
+    call jobsend(getbufvar(s:repl_bufnr, '&channel'), "\<C-c>")
+  else
+    call term_sendkeys(s:repl_bufnr, "\<C-c>")
+    call term_wait(s:repl_bufnr)
+  endif
+endfunction
+
+function! zepl#send_eof() abort
+  if has('nvim')
+    call jobsend(getbufvar(s:repl_bufnr, '&channel'), "\<C-d>")
+  else
+    call term_sendkeys(s:repl_bufnr, "\<C-d>")
+    call term_wait(s:repl_bufnr)
+  endif
+endfunction
+
+function! zepl#send_newline() abort
+  if has('nvim')
+    call jobsend(getbufvar(s:repl_bufnr, '&channel'), "\<CR>")
+  else
+    call term_sendkeys(s:repl_bufnr, "\<CR>")
+    call term_wait(s:repl_bufnr)
+  endif
+endfunction
```

Comment:
These could be simplified.  You mentioned a problem with using `zepl#send` not sending keys I've just pushed some changes which should fix that.  The function now accepts an optional second parameter. If set to `1`, it won't modify the input before sending it to the REPL.  This means that these now works:

```vim
call zepl#send("\<C-d>", 1)
call zepl#send("\<C-c>", 1)
call zepl#send("\<CR>", 1)
```

Reply:
Alright, thanks. One of the problems I was running into with `zepl#send`, and still do even with the verbatim flag, is with mappings.

For example, I have a [`<Plug>` mapping](https://github.com/thezeroalpha/zepl.vim/blob/81a087680c62c878214a2ad9a6e5008a83464291/plugin/zepl.vim#L15) for ctrl-c, and then [a normal mode mapping](https://github.com/thezeroalpha/zepl.vim/blob/81a087680c62c878214a2ad9a6e5008a83464291/plugin/zepl.vim#L26) that runs the `<Plug>` mapping. When the mapping uses `zepl#send` with the verbatim flag, it doesn't work, whereas when there's a separate function just to send control-c, it works. This is the same for sending a newline, except the control-c mapping doesn't give an error message, while the newline mapping says 'missing quote: \\", invalid arguments for function zepl#send'. Only the control-d mapping works with `zepl#send`.

This is the reason I originally wrote three separate functions, because it works. But it's ugly, since it requires three functions that do almost the same thing. I couldn't figure out what was going wrong, do you have any ideas for what might be causing the error?